src/journal/journald-syslog.c

```diff
@@ -371,6 +372,9 @@ void server_process_syslog_message(
         if (!client_context_test_priority(context, priority))
                 return;
 
+        if (client_context_check_keep_log(context, msg, strlen(msg)) <= 0)
```

Comment:
I think `buf` and `raw_len` should be passed here instead of `msg` and `strlen(msg)`.

Reply:
AFAICT `buf` also contains metadata (log level for example). This would create a discrepancy between `journald-syslog.c` and `journald-native.c`, as we filter only on the message.

Reply:
That's a good point. Looking at
```sh
journalctl _PID=1604344 --no-pager --output json-pretty --output-fields MESSAGE,SYSLOG_RAW
{
	"__REALTIME_TIMESTAMP" : "1658940147271764",
	"_BOOT_ID" : "8a9386c8c72f48aebe287664d9a67231",
	"MESSAGE" : "test",
	"SYSLOG_RAW" : [ # this is "<1>test\0me\n"
		60,
		49,
		62,
		116,
		101,
		115,
		116,
		0,
		109,
		101,
		10
	],
	"__CURSOR" : "s=b80498ffccf64015862774053f420ecd;i=6511b0;b=8a9386c8c72f48aebe287664d9a67231;m=50f57a1cb0;t=5e4cc1a2e8854;x=baffc2dab46d74d2",
	"__MONOTONIC_TIMESTAMP" : "347715804336"
}
```
I agree that it doesn't make sense to look at the first three bytes in `RAW_SYSLOG` but I think it would be useful to be able to catch "fishy" messages with unescaped binary data and `strlen(msg)` seems to prevent that.

Reply:
I agree, although until `journald-native.c` is able to do it too, I would prefer to keep it this way for consistency purpose.